pkg/controller/controller_binding.go

```diff
@@ -349,38 +353,75 @@ func isPlanBindable(serviceClass *v1alpha1.ServiceClass, plan *v1alpha1.ServiceP
 
 func (c *controller) injectBinding(binding *v1alpha1.Binding, credentials map[string]interface{}) error {
 	glog.V(5).Infof("Creating Secret %v/%v", binding.Namespace, binding.Spec.SecretName)
-	secret := &v1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      binding.Spec.SecretName,
-			Namespace: binding.Namespace,
-		},
-		Data: make(map[string][]byte),
-	}
 
+	secretData := make(map[string][]byte)
 	for k, v := range credentials {
 		var err error
-		secret.Data[k], err = serialize(v)
+		secretData[k], err = serialize(v)
 		if err != nil {
 			return fmt.Errorf("Unable to serialize credential value %q: %v; %s",
 				k, v, err)
 		}
 	}
 
-	found := false
+	secretClient := c.kubeClient.Core().Secrets(binding.Namespace)
+	// Retry loop for creating/updating the Secret
+	err := wait.PollImmediate(1*time.Second, 1*time.Minute, func() (bool, error) {
```

Comment:
@pmorie that is what we do in Smith - we don't retry, instead we rely on events from concurrent update/create/delete to trigger reconciliation.
I'm not familiar enough with svc-cat codebase to have an opinion in this case. Retrying should be safe but ideally it should not be necessary, I agree.
I found `"k8s.io/kubernetes/pkg/client/retry" RetryOnConflict()` which is built to do this. If you search for usages of this method you'll find similar places in k8s codebase.

I think we need to have the big picture documented - on what events to what object kinds should Binding/Instance/ServiceClass/Broker be reconciled.